hudi-cli/src/main/java/org/apache/hudi/cli/commands/SparkMain.java

```diff
@@ -133,6 +134,13 @@ String getPropsFilePath(String[] args) {
     }
   }
 
+  public static void addAppArgs(SparkLauncher sparkLauncher, SparkMain.SparkCommand cmd, String... args) {
+    //cmd is going to be the first arg so that is why it is minArgsCount - 1
+    ValidationUtils.checkArgument(args.length == cmd.minArgsCount - 1, "App args does not match minArgsCount");
```

Comment:
Could you make the error message easier to understand?  User may not understand what `App args` and `minArgsCount` are without reading the code.

Reply:
The user should never see this. It should only fail if a dev changes the arg list but doesn't modify the enum

Reply:
Then add a prefix: `For developers only: ...`